documentation/src/main/asciidoc/mapstruct-reference-guide.asciidoc

```diff
@@ -1974,3 +1974,139 @@ The order in which the selected methods are applied is roughly determined by the
 ====
 `@BeforeMapping` and `@AfterMapping` are considered experimental as of the 1.0.0.CR1 release. Details in the selection of before/after mapping methods that are applicable for a mapping method or the order in which they are called might still be changed.
 ====
+[[using-spi]]
+== Using the MapStruct SPI
+=== Custom Accessor Naming Strategy
+
+MapStruct offers the possibility to override the `AccessorNamingStrategy` via the SPI interface. A nice example is the use of the fluent API on the source object `GolfPlayer` and `GolfPlayerDto` below.
+
+.Source object with fluent API.
+====
+[source, java, linenums]
+[subs="verbatim,attributes"]
+----
+public class GolfPlayerDto {
+
+    private double handicap;
+    private String name;
+
+    public double handicap() {
+        return handicap;
+    }
+
+    public void withHandicap(double handicap) {
+        this.handicap = handicap;
+    }
+
+    public String name() {
+        return name;
+    }
+
+    public void withName(String name) {
+        this.name = name;
+    }
+----
+====
+
+We want `GolfPlayer` to be mapped to a target object `GolfPlayerDto` similar like we 'always' do this:
+
+.Source object with fluent API.
+====
+[source, java, linenums]
+[subs="verbatim,attributes"]
+----
+@Mapper
+public interface GolfPlayerMapper {
+
+    GolfPlayerMapper INSTANCE = Mappers.getMapper( GolfPlayerMapper.class );
+
+    GolfPlayerDto golfPlayerToDto(GolfPlayer player);
+}
+----
+====
+
+This can be achieved with implementing a ServiceProviderInterface (SPI). Here's an implemented `org.mapstruct.ap.spi.AccessorNamingStrategy`:
+
+.CustomAccessorNamingStrategy
+====
+[source, java, linenums]
+[subs="verbatim,attributes"]
+----
+public class CustomAccessorNamingStrategy implements AccessorNamingStrategy {
```

Comment:
done.. But I needed to adapt the `DefaultAccessorNamingStrategy`  see PR #866 as well.